core/src/main/java/com/exadel/etoolbox/querykit/core/models/query/QueryStatisticModel.java

```diff
@@ -0,0 +1,256 @@
+    private ResourceResolver resourceResolver;
+
+    private List<QueryStatisticModel.QueryInfo> allQueries;
+    private Map<String, Integer> indexesByUsage = new LinkedHashMap<>();
+    private Map<String, List<QueryStatisticModel.QueryInfo>> queriesByIndex = new HashMap<>();
+    private LinkedHashMap<String, List<QueryStatisticModel.QueryInfo>> queriesByIndexSorted = new LinkedHashMap<>();
+
+    @PostConstruct
+    protected void init() throws IOException {
+
+        QueryManager queryManager = getQueryManager(resourceResolver);
+        MBeanServerConnection server = ManagementFactory.getPlatformMBeanServer();
+
+        allQueries = getQueryInfo(server, queryManager);
+
+        for (QueryStatisticModel.QueryInfo queryInfo : allQueries) {
+            if (queryInfo.getIndexes().isEmpty()) {
+                continue;
+            }
+            queryInfo.getIndexes().forEach(index -> queriesByIndex.computeIfAbsent(index, k -> new ArrayList<>()).add(queryInfo));
+        }
+
+        queriesByIndex.entrySet().stream()
+                .sorted((a, b) -> Integer.compare(a.getValue().size(), b.getValue().size()) * -1)
+                .forEach(entry -> queriesByIndexSorted.put(entry.getKey(), entry.getValue()));
+
+        getIndexes(resourceResolver).forEach(index -> indexesByUsage.put(
+                index,
+                queriesByIndex.entrySet().stream()
+                        .filter(entry -> {
+                            String indexId = getIndexId(index);
+                            return entry.getKey().equals(indexId) || entry.getKey().endsWith(indexId + ")");
+                        })
+                        .map(entry -> entry.getValue().size())
+                        .findFirst().orElse(0)));
+    }
+
+    private QueryManager getQueryManager(ResourceResolver resourceResolver) {
+        Session session = resourceResolver.adaptTo(Session.class);
+        if (session == null) {
+            return null;
+        }
+        try {
+            return session.getWorkspace().getQueryManager();
+        } catch (RepositoryException e) {
+            return null;
+        }
+    }
+
+    private List<QueryStatisticModel.QueryInfo> getQueryInfo(MBeanServerConnection server, QueryManager queryManager) {
+        ObjectName queryStatMbean = getQueryStatMBean(server);
+        if (queryStatMbean == null || queryManager == null) {
+            return Collections.emptyList();
+        }
+        String jsonRaw;
+        List<QueryStatisticModel.QueryInfo> result = new ArrayList<>();
+
+        try {
+            jsonRaw = server.invoke(queryStatMbean, "asJson", null, null).toString();
+
+        } catch (Exception e) {
+            return Collections.emptyList();
+        }
+
+        JsonParser jsonParser = new JsonParser();
+        JsonArray queries = jsonParser.parse(jsonRaw).getAsJsonArray();
+
+        for(int i = 0; i < queries.size(); i++) {
+            JsonObject jsonObject = queries.get(i).getAsJsonObject();
```

Comment:
The content of this loop is kind of asking for extracting into a separate method)

Reply:
Exatracted)